lib/awful/widget/layoutlist.lua

```diff
@@ -0,0 +1,442 @@
+        local icn_path, icon = beautiful["layout_" .. (l.name or "")]
+        if icn_path then
+            icon = surface.load(icn_path)
+        end
+
+        table.insert(self._private.cache, {
+            icon         = show_icon and icon   or nil,
+            name         = show_text and l.name or nil,
+            layout       = l,
+            screen       = s,
+            callback     = function() alayout.set(l) end,
+            style        = self._private.style,
+        })
+    end
+end
+
+local function update(self)
+    assert(self._private.layout)
+    awcommon.list_update(
+        self._private.layout,
+        self._private.buttons or default_buttons,
+        wb_label,
+        self._private.data,
+        self._private.cache,
+        {
+            widget_template = self._private.widget_template
+        }
+    )
+end
+
+local layoutlist = {}
+
+--- The layoutlist default widget layout.
+-- If no layout is specified, a `wibox.layout.fixed.vertical` will be created
+-- automatically.
+-- @property base_layout
+-- @param widget
+-- @see wibox.layout.fixed.vertical
+-- @see base_layout
+
+--- The delegate widget template.
+-- @property widget_template
+-- @param table
+
+--- The layoutlist screen.
+-- @property screen
+-- @param screen
+
+--- A function that returns the list of layout to display.
+--
+-- @property source
+-- @param[opt=awful.widget.layoutlist.source.for_screen] function
+
+--- The layoutlist filter function.
+-- @property filter
+-- @param[opt=awful.widget.layoutlist.source.for_screen] function
+
+--- The layoutlist buttons.
+--
+-- The default is:
+--
+--    gears.table.join(
+--        awful.button({ }, 1, awful.layout.set)
+--    )
+--
+-- @property buttons
+-- @param table
+
+--- The default foreground (text) color.
+-- @beautiful beautiful.layoutlist_fg_normal
+-- @tparam[opt=nil] string|pattern fg_normal
+-- @see gears.color
+
+--- The default background color.
+-- @beautiful beautiful.layoutlist_bg_normal
+-- @tparam[opt=nil] string|pattern bg_normal
+-- @see gears.color
+
+--- The selected layout foreground (text) color.
+-- @beautiful beautiful.layoutlist_fg_selected
+-- @tparam[opt=nil] string|pattern fg_selected
+-- @see gears.color
+
+--- The selected layout background color.
+-- @beautiful beautiful.layoutlist_bg_selected
+-- @tparam[opt=nil] string|pattern bg_selected
+-- @see gears.color
+
+--- Disable the layout icons (only show the name label).
+-- @beautiful beautiful.layoutlist_disable_icon
+-- @tparam[opt=false] boolean layoutlist_disable_icon
+
+--- Disable the layout name label (only show the icon).
+-- @beautiful beautiful.layoutlist_disable_name
+-- @tparam[opt=false] boolean layoutlist_disable_name
+
+--- The layoutlist font.
+-- @beautiful beautiful.layoutlist_font
+-- @tparam[opt=nil] string font
+
+--- The selected layout alignment.
+-- @beautiful beautiful.layoutlist_align
+-- @tparam[opt=left] string align *left*, *right* or *center*
+
+--- The selected layout title alignment.
```

Comment:
This description doesn't match the parameter name.

Reply:
Done